pep-0458.txt

```diff
@@ -82,35 +82,35 @@ possibility of security breaches and prepare PyPI accordingly because it is a
 valuable resource used by thousands, if not millions, of people.
 
 Before the wiki attack, PyPI used MD5 hashes to tell package managers, such as
-pip, whether or not a package was corrupted in transit.  However, the absence
+pip, whether or not a distribution file was corrupted in transit.  However, the absence
 of SSL made it hard for package managers to verify transport integrity to PyPI.
 It was therefore easy to launch a man-in-the-middle attack between pip and
-PyPI, and change package content arbitrarily.  As a result, users could be tricked into
-installing malicious packages.  After the wiki
+PyPI, and change distribution content arbitrarily.  As a result, users could be tricked into
+installing malicious distributions.  After the wiki
 attack, several steps were proposed (some of which were implemented) to deliver
 a much higher level of security than was previously the case. These steps included
 requiring SSL to
 communicate with PyPI [6]_, restricting project names [7]_, and migrating from
 MD5 to SHA-2 hashes [8]_.
 
-Though necessary, these steps are insufficient to protect packages because attacks are still
+Though necessary, these steps are insufficient to protect distributions because attacks are still
 possible through other avenues.  For example, a public mirror is trusted to
 honestly mirror PyPI, but some mirrors may misbehave, whether by accident or through
 malicious intervention.
 Package managers such as pip are supposed to use signatures from PyPI to verify
-packages downloaded from a public mirror [9]_, but none are known to actually
+distribution files downloaded from a public mirror [9]_, but none are known to actually
 do so [10]_.  Therefore, it would be wise to add more security measures to
 detect attacks from public mirrors or content delivery networks [11]_ (CDNs).
 
 Even though official mirrors have been deprecated on PyPI [12]_, a
 wide variety of other attack vectors on package managers remain [13]_.  These attacks
-can crash client systems, cause obsolete packages to be installed, or even
+can crash client systems, cause obsolete distributions to be installed, or even
 allow an attacker to execute arbitrary code.  In `September 2013`__, a post was
 made to the Distutils mailing list showing that the latest version of pip (at
 the time) was susceptible to such attacks, and how TUF could protect users
 against them [14]_.  Specifically, testing was done to see how pip would
 respond to these attacks with and without TUF.  Attacks tested included replay
-and freeze, arbitrary packages, slow retrieval, and endless data.  The post
+and freeze, arbitrary distribution, slow retrieval, and endless data.  The post
```

Comment:
isn't "arbitrary packages" a name/class of attack from the relevant paper? If not several of these attack names might want to be made more descriptive to the reader

```suggestion
and freeze, arbitrary distribution files, slow retrieval, and endless data.  The post
```

Reply:
You're right, "A look in the mirror" calls this attack "arbitrary package", but the spec calls it "arbitrary installation", and the reference implementation "arbitrary software installation". Let's use the spec term. :)